xtable-service/src/main/java/org/apache/xtable/service/ConversionService.java

```diff
@@ -0,0 +1,192 @@
+    this.conversionController = new ConversionController(hadoopConf);
+  }
+
+  private Configuration getHadoopConf() {
+    Configuration conf = new Configuration();
+    String hadoopConfigPath = serviceConfig.getHadoopConfigPath();
+    try {
+      // Load configuration from the specified XML file
+      conf.addResource(hadoopConfigPath);
+
+      // If the resource wasn’t found, log a warning
+      if (conf.size() == 0) {
+        log.warn(
+            "Could not load Hadoop configuration from: {}. Using default Hadoop configuration.",
+            hadoopConfigPath);
+      }
+    } catch (Exception e) {
+      log.error(
+          "Error loading Hadoop configuration from: {}. Exception: {}",
+          hadoopConfigPath,
+          e.getMessage(),
+          e);
+    }
+    return conf;
+  }
+
+  /**
+   * Constructs a ConversionService instance using dependency injection for testing.
+   *
+   * @param conversionController a preconfigured conversion controller
+   */
+  @VisibleForTesting
+  public ConversionService(
+      ConversionServiceConfig serviceConfig,
+      ConversionController conversionController,
+      Configuration hadoopConf) {
+    this.serviceConfig = serviceConfig;
+    this.conversionController = conversionController;
+    this.hadoopConf = hadoopConf;
+  }
+
+  /**
+   * Converts a source table to one or more target table formats.
+   *
+   * <p>The method builds a SourceTable based on the request parameters and constructs corresponding
+   * TargetTable instances for each target format. It then performs a synchronous conversion through
+   * the conversion controller. After conversion, it retrieves schema and metadata paths for each
+   * target table.
+   *
+   * @param convertTableRequest the conversion request containing source table details and target
+   *     formats
+   * @return a response containing details of converted target tables
+   */
+  public ConvertTableResponse convertTable(ConvertTableRequest convertTableRequest) {
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(convertTableRequest.getSourceTableName())
+            .basePath(convertTableRequest.getSourceTablePath())
+            .formatName(convertTableRequest.getSourceFormat())
+            .build();
+
+    List<TargetTable> targetTables = new ArrayList<>();
+    for (String targetFormat : convertTableRequest.getTargetFormats()) {
+      TargetTable targetTable =
+          TargetTable.builder()
+              .name(convertTableRequest.getSourceTableName())
+              .basePath(convertTableRequest.getSourceTablePath())
+              .formatName(targetFormat)
+              .build();
+      targetTables.add(targetTable);
+    }
+
+    ConversionConfig conversionConfig =
+        ConversionConfig.builder().sourceTable(sourceTable).targetTables(targetTables).build();
+
+    conversionController.sync(
+        conversionConfig,
+        ConversionUtils.getConversionSourceProvider(
```

Comment:
Why do we initialize a new provider for every request and not during the server bootstrap?

Reply:
@vinishjail97 How do we know which conversion provider to initialize during bootstrap though, the request will tell us which source format provider and which target format provider we need.

Unless you think it might be better to just have all three formats initialized, prior to the request coming, and then we just use the corresponding one.

Reply:
We can either do it in server bootstrap or do it only once per table format but not for every request.